plaidbench.py

```diff
@@ -92,136 +188,193 @@ def main():
-        x_train = np.load(cifar_path).repeat(1 + batch_size/16, axis=0)[:batch_size]
-        y_train_cats = None
+    epoch_size = examples / epochs
 
+    # Stopwatch and Output intialization
     stop_watch = StopWatch(args.callgrind)
     compile_stop_watch = StopWatch(args.callgrind)
     output = Output()
-    data = {
-        'example': args.module
-    }
-    stop_watch.start_outer()
-    compile_stop_watch.start_outer()
-    try:
-        this_dir = os.path.dirname(os.path.abspath(__file__))
-        module = os.path.join(this_dir, 'networks', '%s.py' % args.module)
-        globals = {}
-        exec_(open(module).read(), globals)
-
-        x_train = globals['scale_dataset'](x_train)
-
-        model = globals['build_model']()
-        print("\nModel loaded.")
-
-        # Prep the model and run an initial un-timed batch
-        print("Compiling and running initial batch, batch_size={}".format(batch_size))
-        compile_stop_watch.start()
-        optimizer = 'sgd'
-        if args.module[:3] == 'vgg':
-            from keras.optimizers import SGD
-            optimizer = SGD(lr=0.0001)
-        model.compile(optimizer=optimizer, loss='categorical_crossentropy',
-                      metrics=['accuracy'])
-
-        if args.train:
-            # training
-            x = x_train[:epoch_size]
-            y = y_train[:epoch_size]
-            model.train_on_batch(x_train[0:batch_size], y_train[0:batch_size])
-            compile_stop_watch.stop()
-            for i in range(args.epochs):
-                if i == 1:
-                    print('Doing the main timing')
-                stop_watch.start()
-                history = model.fit(x=x, y=y, batch_size=batch_size, epochs=1, shuffle=False, initial_epoch=0)
-                stop_watch.stop()
-                time.sleep(.025 * random.random())
-                if i == 0:
-                    output.contents = [history.history['loss']]
-            output.contents = np.array(output.contents)
-        else:
-            # inference
-            y = model.predict(x=x_train, batch_size=batch_size)
-            compile_stop_watch.stop()
-            output.contents = y
-            print('Warmup')
- 
-            for i in range(32//batch_size + 1):
-                y = model.predict(x=x_train, batch_size=batch_size)
-            # Now start the clock and run 100 batches
-            print('Doing the main timing')
+    data['train'] = args.train
 
-            for i in range(examples//batch_size):
-                stop_watch.start()
-                y = model.predict(x=x_train, batch_size=batch_size)
-                stop_watch.stop()
-                time.sleep(.025 * random.random())
+    # Blanket run - runs every supported network
+    if args.module == 'blanket':
+        data['blanket_run'] = True
+        outputs = {}
+        outputs['run_configuration'] = data.copy()
+        
+        print("Running blanket, setting examples size to 256 for speediness")
```

Comment:
only set to 256 if the user didn't explicitly set the parameter -- I would make the examples parameter default to None, and then if it's None, set it to 256 if blanket, otherwise 1024